### PR TITLE
Remove BundlerMinifier in bundling-and-minification-md

### LINKED ISSUE #21659: Remove B&M for all versions
See internal email. Subject: **Bundle and minify**

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 36f55e9a-cd3f-83a0-f142-969dc10d3ceb
* Version Independent ID: f6845f70-1ac6-2036-f213-a3b665b3d8bf
* Content: [Bundle and minify static assets in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/client-side/bundling-and-minification?view=aspnetcore-5.0)
* Content Source: [aspnetcore/client-side/bundling-and-minification.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/client-side/bundling-and-minification.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-clientside**
* GitHub Login: @scottaddie
* Microsoft Alias: **scaddie**

### PR DESCRIPTION
**Pull request by wadepickett**
Fixes #21659

[Internal review page](https://review.docs.microsoft.com/en-us/aspnet/core/client-side/bundling-and-minification?branch=pr-en-us-21770)

### CODE REVIEW

File: aspnetcore/client-side/bundling-and-minification.md [hunk 1]
```diff
@@ -54,16 +54,15 @@ File Requests  | 7   | 18     | 157%
...
+ASP.NET Core is compatible with WebOptimizer, an open-source bundling and minification solution. For instruction and sample projects using WebOptimizer, see [WebOptimizer](https://github.com/ligershark/WebOptimizer). ASP.NET Core does not provide a native bundling and minification solution.
 
-## Configure bundling and minification
+The MVC and Razor Pages project templates provide a configuration solution for bundling and minification workflow consisting of a JSON configuration file. Third-party tools, such as the [Grunt](xref:client-side/using-grunt) task runner, accomplish the same tasks with a bit more complexity. A third-party tool is a great fit when your development workflow requires processing beyond bundling and minification&mdash;such as linting and image optimization. By using design-time bundling and minification, the minified files are created prior to the app's deployment. Bundling and minifying before deployment provides the advantage of reduced server load. However, it's important to recognize that design-time bundling and minification increases build complexity and only works with static files.
```
Comment (scottaddie) [inline aspnetcore/client-side/bundling-and-minification.md:63]: Remove this **Configure bundling and minification** section, as it's based on _bundleconfig.json_. That JSON file existed to support Mads' VS extension.

File: aspnetcore/client-side/bundling-and-minification.md [hunk 2]
```diff
@@ -96,7 +95,7 @@ Configuration options include:
 
 ## Add files to workflow
```
Comment (scottaddie) [inline aspnetcore/client-side/bundling-and-minification.md:96]: This section should be removed too, since it's based on _bundleconfig.json_.

**Review by scottaddie — CHANGES_REQUESTED** [review]
In addition to the suggestions below, please remove the **Consume bundleconfig.json from Gulp** section.

**Comment by scottaddie** [comment]
As part of this PR, I suggest we also retire https://docs.microsoft.com/aspnet/core/client-side/using-grunt. Grunt is an open-source tool that Microsoft doesn't own, and this doc is seriously outdated. Grunt was used for bundling & minification is the very early days of ASP.NET Core (in the days of DNX and _project.json_). It too was removed from the project templates. After retiring the doc, you can redirect to https://gruntjs.com.

**Comment by wadepickett** [comment]
@scottaddie, I will remove using-grunt.md as a 2nd pr today since an number of topics reference it, and this needs to be published right away.

**Comment by wadepickett** [comment]
@scottaddie, The gulpfile.js example we specify `bundleconfig = require('./bundleconfig.json');`     The json could be named anything in this case, and it makes sense to call your own bundleconfig.json, but that might also cause some confusion since the template used to provide that.  I'm thinking we should just leave it named thus however.

**Comment by scottaddie** [comment]
@wadepickett Let's remove that entire **Consume bundleconfig.json from Gulp** section. The VS extension provided an option to right-click the *bundleconfig.json* file and convert it to *gulpfile.js*. We don't want to lead anyone to believe that's still supported.

**Comment by wadepickett** [comment]
@scottaddie, per our discussion I will remove the sample folder plus these sections in moment.
•	Consume bundleconfig.json from Gulp
•	Run Gulp tasks

**Comment by wadepickett** [comment]
@madskristensen, for visibility, review.

File: aspnetcore/client-side/bundling-and-minification.md [hunk 3]
```diff
@@ -54,66 +67,13 @@ File Requests  | 7   | 18     | 157%
...
-> Alternatively, the following globbing pattern could be used:
->
-> ```json
-> "inputFiles": ["wwwroot/**/!(*.min).css" ]
-> ```
->
-> This globbing pattern matches all CSS files and excludes the minified file pattern.
-
-Build the application. Open *site.min.css* and notice the content of *custom.css* is appended to the end of the file.
+Third-party tools, such as the [Grunt](https://gruntjs.com) task runner, provide a workflow solution for bundling and minification, as well as linting and image optimization. By using design-time bundling and minification, the minified files are created prior to the app's deployment. Bundling and minifying before deployment provides the advantage of reduced server load. However, it's important to recognize that design-time bundling and minification increases build complexity and only works with static files.
```
Comment (scottaddie) [inline aspnetcore/client-side/bundling-and-minification.md:76]:
```suggestion
Third-party tools, such as [Gulp](https://gulpjs.com) and [Webpack](https://webpack.js.org), provide workflow automation for bundling and minification, as well as linting and image optimization. By using design-time bundling and minification, the minified files are created prior to the app's deployment. Bundling and minifying before deployment provides the advantage of reduced server load. However, it's important to recognize that design-time bundling and minification increases build complexity and only works with static files.
```

**Review by scottaddie — CHANGES_REQUESTED** [review]
See my suggestions below. Also, I recommend removing the **Use Grunt** link from the **Additional resources** section. Grunt was used in the project templates only in the very early days of ASP.NET Core. It was later replaced with Gulp.

**Comment by madskristensen** [comment]
LGTM

**Comment by wadepickett** [comment]
Updates per review all in, however OPS services are not available as of a few minutes ago, so I can't get a doc validation and build until that is fixed.  I'm watching for it...

**Comment by scottaddie** [comment]
@wadepickett The doc build failure is likely related to the current Azure AD outage. Try retriggering the build once that outage is resolved.

**Review by scottaddie — APPROVED** [review]

**Pull request merged**